Feature/issue#18 lic13

ErikBetzholtz opened the pull request:
Closes #18 
will add LIC13 function and related tests. Does not solve problem of EvaluateLICTeast file containing duplicate tests for LIC6

ErikBetzholtz committed:
feat: add first version of LIC13 function

This commit will add the first version of the LIC13 function without tests

ErikBetzholtz committed:
feat: add LIC13 function and related tests

this commit will add the LIC13 function with in-code documentation as well as four associated tests to LIC13

ErikBetzholtz committed:
minor: temporary remove LIC13 tests for local merge

will temporarily remove tests for LIC13 in order to resolve merge conflict when merging main into current branch locally

ErikBetzholtz committed:
Squashed commit of the following:

commit cab982a49d3d4683e7b6498a551b022f67742940
Merge: 7dcb548 7fe141c
Author: Vincent Lagerros <36774549+VincentLagerros@users.noreply.github.com>
Date:   Wed Jan 29 20:50:16 2025 +0100

    Merge pull request #42 from VincentLagerros/LIC14

    feat: Added LIC14

commit 7fe141c3d4129fb89343fbeebc4aa81b55d0652e
Merge: c70068d 7dcb548
Author: Vincent Lagerros <36774549+VincentLagerros@users.noreply.github.com>
Date:   Wed Jan 29 18:09:15 2025 +0100

    Merge branch 'main' into LIC14

commit c70068d71aa03804decc44d004110872e0b3da08
Author: Vincent <vincent.lagerros@gmail.com>
Date:   Wed Jan 29 18:05:19 2025 +0100

    feat: Added LIC14

    Closes #14, adds trianglearea and fixes formating

commit 7dcb548f821bd18ee43f2122d2b4240540795a7f
Merge: fb6fdd8 a803077
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Wed Jan 29 17:48:28 2025 +0100

    Merge pull request #41 from VincentLagerros/LIC1

    feat: add LIC1 and connected tests

commit a803077c16dcfd9c88b140974fe7787666acd2b9
Author: Julia Potrus <juliapo@kth.se>
Date:   Wed Jan 29 17:46:33 2025 +0100

    fix: add case where points on single line exceed radius1*2, and test for this

commit fb6fdd861b4181246933f40d566b2a82d0f32fd0
Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
Date:   Wed Jan 29 17:29:25 2025 +0100

    feat: add LIC4 and related tests (#32)

    * feat: add LIC4 funciton and related tests

    This will add the LIC4 function and related tests

    * doc: add in-code documentation for LIC4

    commit will add in-code documentation for LIC4 function

    * test: add exception tests for LIC4

    will add tests for invalid input to the LIC4 function to assure exceptions are thrown

    * minor: change order of tests in code

    will change the order of tests in the EvaluateLICTest file to be in order

    * Squashed commit of the following:

    commit 73c6bb786f00ea700d05d0a703cf322282bb16f2
    Merge: bfe99cb 5834072
    Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
    Date:   Wed Jan 29 16:41:27 2025 +0100

        Merge pull request #40 from VincentLagerros/lic9bugfix

        Bugfix: fix LIC9 by changing

    commit 5834072b7ced9aec42e97f8e5bbfdfa120e12123
    Author: Vincent <vincent.lagerros@gmail.com>
    Date:   Wed Jan 29 16:20:34 2025 +0100

        Bugfix: fix LIC9 by changing

        Fixes Consecutive data points interpretation

    commit bfe99cb6cd6d2c9b2302dcaf700f0cb6b444aba6
    Merge: 8f252dd 00dc2b2
    Author: RasmussBrodin <126715286+RasmussBrodin@users.noreply.github.com>
    Date:   Wed Jan 29 16:13:49 2025 +0100

        Merge pull request #38 from VincentLagerros/Fix/isssue#37

        fix: Added input test for LIC6

    commit 00dc2b217d991e0a65aebbc03f2231d4e800db68
    Merge: d1a9201 8f252dd
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 16:12:55 2025 +0100

        Merge branch 'main' into Fix/isssue#37

    commit 8f252dd17e822f39f2a262acd37419745f48129d
    Merge: 4aee3e1 ff82cc6
    Author: RasmussBrodin <126715286+RasmussBrodin@users.noreply.github.com>
    Date:   Wed Jan 29 16:01:52 2025 +0100

        Merge pull request #33 from VincentLagerros/Feture/issue#9

        feat: Added LIC7 and tests

    commit ff82cc631fa7ebe74afdfd05d679ada2cc4fd9ff
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 15:57:18 2025 +0100

        refactor: LIC7 now uses distance()

        I did not know about distance(). The code is much cleaner now.

        Thanks!

    commit 4aee3e1e9244d5805d0d14c889047c0dc152e0d7
    Merge: 8238a3d 858b765
    Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
    Date:   Wed Jan 29 15:55:21 2025 +0100

        Merge pull request #34 from VincentLagerros/feature/issue#6

        feat: Added LIC2 and test cases

    commit d1a9201056fa87ce5b07c51a5b905df6b3719890
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 14:24:59 2025 +0100

        fix: Added input test for LIC6

        This closes issue #6.

        No change in code, only added test cases and it passes so should be able to merge without any problems.

    commit 858b765ab45ba4b0009d10f8f8cb2a5b5bfd4c6e
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 12:23:53 2025 +0100

        Fix: Added test cases for input validation

    commit 77772236a0cb77f25f9de9e788d7a52f5c9bfdc1
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 12:18:03 2025 +0100

        Fix: Added input validating test cases

    commit 5d6574f41badf5ae6a5f7591d123d6b94f43a9f3
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 01:11:10 2025 +0100

        feat: Added LIC2 and test cases

        Im unsure if these test cases are sufficient, but they test the fundamental functionality

    commit dd7ce34616a805e45098376106a35e975f4171c6
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 00:51:00 2025 +0100

        feat: Added LIC7 and tests

    * refactor: change LIC4 to be more memory efficient

    This commit will change LIC4 to use an array of size clusterSize to keep track of unique quadrants of coordinates in the moving window of coordinates. This instead of using an ArrayList of precomputed coordinates and a HashSet for checking for number of unique coordinates in window

commit 8605e213027a727a9d4b913f61e60c7fc4510e2b
Merge: 73c6bb7 7474a57
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Wed Jan 29 17:24:14 2025 +0100

    Merge pull request #35 from VincentLagerros/Feture/issue#15

    feat: Added LIC10, LIC11 and tests

commit 15c4b30d81305bd2d60dce987bd35ffe4b99f3f9
Merge: 23faf9c 73c6bb7
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Wed Jan 29 17:22:56 2025 +0100

    Merge branch 'main' into LIC1

commit 23faf9c1ebd92f936e3affc2388f45342ca4b08e
Author: Julia Potrus <juliapo@kth.se>
Date:   Wed Jan 29 17:17:53 2025 +0100

    feat: add LIC1 and connected tests

commit 73c6bb786f00ea700d05d0a703cf322282bb16f2
Merge: bfe99cb 5834072
Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
Date:   Wed Jan 29 16:41:27 2025 +0100

    Merge pull request #40 from VincentLagerros/lic9bugfix

    Bugfix: fix LIC9 by changing

commit 5834072b7ced9aec42e97f8e5bbfdfa120e12123
Author: Vincent <vincent.lagerros@gmail.com>
Date:   Wed Jan 29 16:20:34 2025 +0100

    Bugfix: fix LIC9 by changing

    Fixes Consecutive data points interpretation

commit 7474a574df8ace04a1c73bb60148414c9d584780
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 12:47:05 2025 +0100

    Fix: added input testing to LIC 11

commit e848cddab9933883f39f91b422c594c59caf16b4
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 12:37:57 2025 +0100

    Fix: Added input testing to LIC 10

commit 7e1ce822977f471e5b111ce228efebe1c4df8287
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 11:00:38 2025 +0100

    refactor: changed name of a test for LIC11

commit 2f655f65b24a4c2689c2a2c2965fd9f2f1dbe892
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 10:56:53 2025 +0100

    feat: Added LIC11 and tests

commit ed4650634306def3dfc1d929efabf0f7ed364b8b
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 10:08:37 2025 +0100

    feat: Added LIC10 and tests

ErikBetzholtz committed:
minor: re-add LIC13 tests

will re-add the temporarily removed LIC13 tests to the test file after completed local merge

ErikBetzholtz committed:
Squashed commit of the following:

commit cab982a49d3d4683e7b6498a551b022f67742940
Merge: 7dcb548 7fe141c
Author: Vincent Lagerros <36774549+VincentLagerros@users.noreply.github.com>
Date:   Wed Jan 29 20:50:16 2025 +0100

    Merge pull request #42 from VincentLagerros/LIC14

    feat: Added LIC14

commit 7fe141c3d4129fb89343fbeebc4aa81b55d0652e
Merge: c70068d 7dcb548
Author: Vincent Lagerros <36774549+VincentLagerros@users.noreply.github.com>
Date:   Wed Jan 29 18:09:15 2025 +0100

    Merge branch 'main' into LIC14

commit c70068d71aa03804decc44d004110872e0b3da08
Author: Vincent <vincent.lagerros@gmail.com>
Date:   Wed Jan 29 18:05:19 2025 +0100

    feat: Added LIC14

    Closes #14, adds trianglearea and fixes formating

commit 7dcb548f821bd18ee43f2122d2b4240540795a7f
Merge: fb6fdd8 a803077
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Wed Jan 29 17:48:28 2025 +0100

    Merge pull request #41 from VincentLagerros/LIC1

    feat: add LIC1 and connected tests

commit a803077c16dcfd9c88b140974fe7787666acd2b9
Author: Julia Potrus <juliapo@kth.se>
Date:   Wed Jan 29 17:46:33 2025 +0100

    fix: add case where points on single line exceed radius1*2, and test for this

commit fb6fdd861b4181246933f40d566b2a82d0f32fd0
Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
Date:   Wed Jan 29 17:29:25 2025 +0100

    feat: add LIC4 and related tests (#32)

    * feat: add LIC4 funciton and related tests

    This will add the LIC4 function and related tests

    * doc: add in-code documentation for LIC4

    commit will add in-code documentation for LIC4 function

    * test: add exception tests for LIC4

    will add tests for invalid input to the LIC4 function to assure exceptions are thrown

    * minor: change order of tests in code

    will change the order of tests in the EvaluateLICTest file to be in order

    * Squashed commit of the following:

    commit 73c6bb786f00ea700d05d0a703cf322282bb16f2
    Merge: bfe99cb 5834072
    Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
    Date:   Wed Jan 29 16:41:27 2025 +0100

        Merge pull request #40 from VincentLagerros/lic9bugfix

        Bugfix: fix LIC9 by changing

    commit 5834072b7ced9aec42e97f8e5bbfdfa120e12123
    Author: Vincent <vincent.lagerros@gmail.com>
    Date:   Wed Jan 29 16:20:34 2025 +0100

        Bugfix: fix LIC9 by changing

        Fixes Consecutive data points interpretation

    commit bfe99cb6cd6d2c9b2302dcaf700f0cb6b444aba6
    Merge: 8f252dd 00dc2b2
    Author: RasmussBrodin <126715286+RasmussBrodin@users.noreply.github.com>
    Date:   Wed Jan 29 16:13:49 2025 +0100

        Merge pull request #38 from VincentLagerros/Fix/isssue#37

        fix: Added input test for LIC6

    commit 00dc2b217d991e0a65aebbc03f2231d4e800db68
    Merge: d1a9201 8f252dd
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 16:12:55 2025 +0100

        Merge branch 'main' into Fix/isssue#37

    commit 8f252dd17e822f39f2a262acd37419745f48129d
    Merge: 4aee3e1 ff82cc6
    Author: RasmussBrodin <126715286+RasmussBrodin@users.noreply.github.com>
    Date:   Wed Jan 29 16:01:52 2025 +0100

        Merge pull request #33 from VincentLagerros/Feture/issue#9

        feat: Added LIC7 and tests

    commit ff82cc631fa7ebe74afdfd05d679ada2cc4fd9ff
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 15:57:18 2025 +0100

        refactor: LIC7 now uses distance()

        I did not know about distance(). The code is much cleaner now.

        Thanks!

    commit 4aee3e1e9244d5805d0d14c889047c0dc152e0d7
    Merge: 8238a3d 858b765
    Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
    Date:   Wed Jan 29 15:55:21 2025 +0100

        Merge pull request #34 from VincentLagerros/feature/issue#6

        feat: Added LIC2 and test cases

    commit d1a9201056fa87ce5b07c51a5b905df6b3719890
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 14:24:59 2025 +0100

        fix: Added input test for LIC6

        This closes issue #6.

        No change in code, only added test cases and it passes so should be able to merge without any problems.

    commit 858b765ab45ba4b0009d10f8f8cb2a5b5bfd4c6e
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 12:23:53 2025 +0100

        Fix: Added test cases for input validation

    commit 77772236a0cb77f25f9de9e788d7a52f5c9bfdc1
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 12:18:03 2025 +0100

        Fix: Added input validating test cases

    commit 5d6574f41badf5ae6a5f7591d123d6b94f43a9f3
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 01:11:10 2025 +0100

        feat: Added LIC2 and test cases

        Im unsure if these test cases are sufficient, but they test the fundamental functionality

    commit dd7ce34616a805e45098376106a35e975f4171c6
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 00:51:00 2025 +0100

        feat: Added LIC7 and tests

    * refactor: change LIC4 to be more memory efficient

    This commit will change LIC4 to use an array of size clusterSize to keep track of unique quadrants of coordinates in the moving window of coordinates. This instead of using an ArrayList of precomputed coordinates and a HashSet for checking for number of unique coordinates in window

commit 8605e213027a727a9d4b913f61e60c7fc4510e2b
Merge: 73c6bb7 7474a57
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Wed Jan 29 17:24:14 2025 +0100

    Merge pull request #35 from VincentLagerros/Feture/issue#15

    feat: Added LIC10, LIC11 and tests

commit 15c4b30d81305bd2d60dce987bd35ffe4b99f3f9
Merge: 23faf9c 73c6bb7
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Wed Jan 29 17:22:56 2025 +0100

    Merge branch 'main' into LIC1

commit 23faf9c1ebd92f936e3affc2388f45342ca4b08e
Author: Julia Potrus <juliapo@kth.se>
Date:   Wed Jan 29 17:17:53 2025 +0100

    feat: add LIC1 and connected tests

commit 73c6bb786f00ea700d05d0a703cf322282bb16f2
Merge: bfe99cb 5834072
Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
Date:   Wed Jan 29 16:41:27 2025 +0100

    Merge pull request #40 from VincentLagerros/lic9bugfix

    Bugfix: fix LIC9 by changing

commit 5834072b7ced9aec42e97f8e5bbfdfa120e12123
Author: Vincent <vincent.lagerros@gmail.com>
Date:   Wed Jan 29 16:20:34 2025 +0100

    Bugfix: fix LIC9 by changing

    Fixes Consecutive data points interpretation

commit 7474a574df8ace04a1c73bb60148414c9d584780
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 12:47:05 2025 +0100

    Fix: added input testing to LIC 11

commit e848cddab9933883f39f91b422c594c59caf16b4
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 12:37:57 2025 +0100

    Fix: Added input testing to LIC 10

commit 7e1ce822977f471e5b111ce228efebe1c4df8287
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 11:00:38 2025 +0100

    refactor: changed name of a test for LIC11

commit 2f655f65b24a4c2689c2a2c2965fd9f2f1dbe892
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 10:56:53 2025 +0100

    feat: Added LIC11 and tests

commit ed4650634306def3dfc1d929efabf0f7ed364b8b
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 10:08:37 2025 +0100

    feat: Added LIC10 and tests

ErikBetzholtz committed:
Squashed commit of the following:

commit cab982a49d3d4683e7b6498a551b022f67742940
Merge: 7dcb548 7fe141c
Author: Vincent Lagerros <36774549+VincentLagerros@users.noreply.github.com>
Date:   Wed Jan 29 20:50:16 2025 +0100

    Merge pull request #42 from VincentLagerros/LIC14

    feat: Added LIC14

commit 7fe141c3d4129fb89343fbeebc4aa81b55d0652e
Merge: c70068d 7dcb548
Author: Vincent Lagerros <36774549+VincentLagerros@users.noreply.github.com>
Date:   Wed Jan 29 18:09:15 2025 +0100

    Merge branch 'main' into LIC14

commit c70068d71aa03804decc44d004110872e0b3da08
Author: Vincent <vincent.lagerros@gmail.com>
Date:   Wed Jan 29 18:05:19 2025 +0100

    feat: Added LIC14

    Closes #14, adds trianglearea and fixes formating

commit 7dcb548f821bd18ee43f2122d2b4240540795a7f
Merge: fb6fdd8 a803077
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Wed Jan 29 17:48:28 2025 +0100

    Merge pull request #41 from VincentLagerros/LIC1

    feat: add LIC1 and connected tests

commit a803077c16dcfd9c88b140974fe7787666acd2b9
Author: Julia Potrus <juliapo@kth.se>
Date:   Wed Jan 29 17:46:33 2025 +0100

    fix: add case where points on single line exceed radius1*2, and test for this

commit fb6fdd861b4181246933f40d566b2a82d0f32fd0
Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
Date:   Wed Jan 29 17:29:25 2025 +0100

    feat: add LIC4 and related tests (#32)

    * feat: add LIC4 funciton and related tests

    This will add the LIC4 function and related tests

    * doc: add in-code documentation for LIC4

    commit will add in-code documentation for LIC4 function

    * test: add exception tests for LIC4

    will add tests for invalid input to the LIC4 function to assure exceptions are thrown

    * minor: change order of tests in code

    will change the order of tests in the EvaluateLICTest file to be in order

    * Squashed commit of the following:

    commit 73c6bb786f00ea700d05d0a703cf322282bb16f2
    Merge: bfe99cb 5834072
    Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
    Date:   Wed Jan 29 16:41:27 2025 +0100

        Merge pull request #40 from VincentLagerros/lic9bugfix

        Bugfix: fix LIC9 by changing

    commit 5834072b7ced9aec42e97f8e5bbfdfa120e12123
    Author: Vincent <vincent.lagerros@gmail.com>
    Date:   Wed Jan 29 16:20:34 2025 +0100

        Bugfix: fix LIC9 by changing

        Fixes Consecutive data points interpretation

    commit bfe99cb6cd6d2c9b2302dcaf700f0cb6b444aba6
    Merge: 8f252dd 00dc2b2
    Author: RasmussBrodin <126715286+RasmussBrodin@users.noreply.github.com>
    Date:   Wed Jan 29 16:13:49 2025 +0100

        Merge pull request #38 from VincentLagerros/Fix/isssue#37

        fix: Added input test for LIC6

    commit 00dc2b217d991e0a65aebbc03f2231d4e800db68
    Merge: d1a9201 8f252dd
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 16:12:55 2025 +0100

        Merge branch 'main' into Fix/isssue#37

    commit 8f252dd17e822f39f2a262acd37419745f48129d
    Merge: 4aee3e1 ff82cc6
    Author: RasmussBrodin <126715286+RasmussBrodin@users.noreply.github.com>
    Date:   Wed Jan 29 16:01:52 2025 +0100

        Merge pull request #33 from VincentLagerros/Feture/issue#9

        feat: Added LIC7 and tests

    commit ff82cc631fa7ebe74afdfd05d679ada2cc4fd9ff
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 15:57:18 2025 +0100

        refactor: LIC7 now uses distance()

        I did not know about distance(). The code is much cleaner now.

        Thanks!

    commit 4aee3e1e9244d5805d0d14c889047c0dc152e0d7
    Merge: 8238a3d 858b765
    Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
    Date:   Wed Jan 29 15:55:21 2025 +0100

        Merge pull request #34 from VincentLagerros/feature/issue#6

        feat: Added LIC2 and test cases

    commit d1a9201056fa87ce5b07c51a5b905df6b3719890
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 14:24:59 2025 +0100

        fix: Added input test for LIC6

        This closes issue #6.

        No change in code, only added test cases and it passes so should be able to merge without any problems.

    commit 858b765ab45ba4b0009d10f8f8cb2a5b5bfd4c6e
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 12:23:53 2025 +0100

        Fix: Added test cases for input validation

    commit 77772236a0cb77f25f9de9e788d7a52f5c9bfdc1
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 12:18:03 2025 +0100

        Fix: Added input validating test cases

    commit 5d6574f41badf5ae6a5f7591d123d6b94f43a9f3
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 01:11:10 2025 +0100

        feat: Added LIC2 and test cases

        Im unsure if these test cases are sufficient, but they test the fundamental functionality

    commit dd7ce34616a805e45098376106a35e975f4171c6
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 00:51:00 2025 +0100

        feat: Added LIC7 and tests

    * refactor: change LIC4 to be more memory efficient

    This commit will change LIC4 to use an array of size clusterSize to keep track of unique quadrants of coordinates in the moving window of coordinates. This instead of using an ArrayList of precomputed coordinates and a HashSet for checking for number of unique coordinates in window

commit 8605e213027a727a9d4b913f61e60c7fc4510e2b
Merge: 73c6bb7 7474a57
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Wed Jan 29 17:24:14 2025 +0100

    Merge pull request #35 from VincentLagerros/Feture/issue#15

    feat: Added LIC10, LIC11 and tests

commit 15c4b30d81305bd2d60dce987bd35ffe4b99f3f9
Merge: 23faf9c 73c6bb7
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Wed Jan 29 17:22:56 2025 +0100

    Merge branch 'main' into LIC1

commit 23faf9c1ebd92f936e3affc2388f45342ca4b08e
Author: Julia Potrus <juliapo@kth.se>
Date:   Wed Jan 29 17:17:53 2025 +0100

    feat: add LIC1 and connected tests

commit 73c6bb786f00ea700d05d0a703cf322282bb16f2
Merge: bfe99cb 5834072
Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
Date:   Wed Jan 29 16:41:27 2025 +0100

    Merge pull request #40 from VincentLagerros/lic9bugfix

    Bugfix: fix LIC9 by changing

commit 5834072b7ced9aec42e97f8e5bbfdfa120e12123
Author: Vincent <vincent.lagerros@gmail.com>
Date:   Wed Jan 29 16:20:34 2025 +0100

    Bugfix: fix LIC9 by changing

    Fixes Consecutive data points interpretation

commit 7474a574df8ace04a1c73bb60148414c9d584780
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 12:47:05 2025 +0100

    Fix: added input testing to LIC 11

commit e848cddab9933883f39f91b422c594c59caf16b4
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 12:37:57 2025 +0100

    Fix: Added input testing to LIC 10

commit 7e1ce822977f471e5b111ce228efebe1c4df8287
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 11:00:38 2025 +0100

    refactor: changed name of a test for LIC11

commit 2f655f65b24a4c2689c2a2c2965fd9f2f1dbe892
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 10:56:53 2025 +0100

    feat: Added LIC11 and tests

commit ed4650634306def3dfc1d929efabf0f7ed364b8b
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 10:08:37 2025 +0100

    feat: Added LIC10 and tests

ErikBetzholtz committed:
minor: remove redundant tests

will remove redundant tests for LIC6 that became duplicated after local merge

ErikBetzholtz committed:
Squashed commit of the following:

commit cab982a49d3d4683e7b6498a551b022f67742940
Merge: 7dcb548 7fe141c
Author: Vincent Lagerros <36774549+VincentLagerros@users.noreply.github.com>
Date:   Wed Jan 29 20:50:16 2025 +0100

    Merge pull request #42 from VincentLagerros/LIC14

    feat: Added LIC14

commit 7fe141c3d4129fb89343fbeebc4aa81b55d0652e
Merge: c70068d 7dcb548
Author: Vincent Lagerros <36774549+VincentLagerros@users.noreply.github.com>
Date:   Wed Jan 29 18:09:15 2025 +0100

    Merge branch 'main' into LIC14

commit c70068d71aa03804decc44d004110872e0b3da08
Author: Vincent <vincent.lagerros@gmail.com>
Date:   Wed Jan 29 18:05:19 2025 +0100

    feat: Added LIC14

    Closes #14, adds trianglearea and fixes formating

commit 7dcb548f821bd18ee43f2122d2b4240540795a7f
Merge: fb6fdd8 a803077
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Wed Jan 29 17:48:28 2025 +0100

    Merge pull request #41 from VincentLagerros/LIC1

    feat: add LIC1 and connected tests

commit a803077c16dcfd9c88b140974fe7787666acd2b9
Author: Julia Potrus <juliapo@kth.se>
Date:   Wed Jan 29 17:46:33 2025 +0100

    fix: add case where points on single line exceed radius1*2, and test for this

commit fb6fdd861b4181246933f40d566b2a82d0f32fd0
Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
Date:   Wed Jan 29 17:29:25 2025 +0100

    feat: add LIC4 and related tests (#32)

    * feat: add LIC4 funciton and related tests

    This will add the LIC4 function and related tests

    * doc: add in-code documentation for LIC4

    commit will add in-code documentation for LIC4 function

    * test: add exception tests for LIC4

    will add tests for invalid input to the LIC4 function to assure exceptions are thrown

    * minor: change order of tests in code

    will change the order of tests in the EvaluateLICTest file to be in order

    * Squashed commit of the following:

    commit 73c6bb786f00ea700d05d0a703cf322282bb16f2
    Merge: bfe99cb 5834072
    Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
    Date:   Wed Jan 29 16:41:27 2025 +0100

        Merge pull request #40 from VincentLagerros/lic9bugfix

        Bugfix: fix LIC9 by changing

    commit 5834072b7ced9aec42e97f8e5bbfdfa120e12123
    Author: Vincent <vincent.lagerros@gmail.com>
    Date:   Wed Jan 29 16:20:34 2025 +0100

        Bugfix: fix LIC9 by changing

        Fixes Consecutive data points interpretation

    commit bfe99cb6cd6d2c9b2302dcaf700f0cb6b444aba6
    Merge: 8f252dd 00dc2b2
    Author: RasmussBrodin <126715286+RasmussBrodin@users.noreply.github.com>
    Date:   Wed Jan 29 16:13:49 2025 +0100

        Merge pull request #38 from VincentLagerros/Fix/isssue#37

        fix: Added input test for LIC6

    commit 00dc2b217d991e0a65aebbc03f2231d4e800db68
    Merge: d1a9201 8f252dd
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 16:12:55 2025 +0100

        Merge branch 'main' into Fix/isssue#37

    commit 8f252dd17e822f39f2a262acd37419745f48129d
    Merge: 4aee3e1 ff82cc6
    Author: RasmussBrodin <126715286+RasmussBrodin@users.noreply.github.com>
    Date:   Wed Jan 29 16:01:52 2025 +0100

        Merge pull request #33 from VincentLagerros/Feture/issue#9

        feat: Added LIC7 and tests

    commit ff82cc631fa7ebe74afdfd05d679ada2cc4fd9ff
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 15:57:18 2025 +0100

        refactor: LIC7 now uses distance()

        I did not know about distance(). The code is much cleaner now.

        Thanks!

    commit 4aee3e1e9244d5805d0d14c889047c0dc152e0d7
    Merge: 8238a3d 858b765
    Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
    Date:   Wed Jan 29 15:55:21 2025 +0100

        Merge pull request #34 from VincentLagerros/feature/issue#6

        feat: Added LIC2 and test cases

    commit d1a9201056fa87ce5b07c51a5b905df6b3719890
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 14:24:59 2025 +0100

        fix: Added input test for LIC6

        This closes issue #6.

        No change in code, only added test cases and it passes so should be able to merge without any problems.

    commit 858b765ab45ba4b0009d10f8f8cb2a5b5bfd4c6e
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 12:23:53 2025 +0100

        Fix: Added test cases for input validation

    commit 77772236a0cb77f25f9de9e788d7a52f5c9bfdc1
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 12:18:03 2025 +0100

        Fix: Added input validating test cases

    commit 5d6574f41badf5ae6a5f7591d123d6b94f43a9f3
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 01:11:10 2025 +0100

        feat: Added LIC2 and test cases

        Im unsure if these test cases are sufficient, but they test the fundamental functionality

    commit dd7ce34616a805e45098376106a35e975f4171c6
    Author: RasmussBrodin <rasmuss.brodin@gmail.com>
    Date:   Wed Jan 29 00:51:00 2025 +0100

        feat: Added LIC7 and tests

    * refactor: change LIC4 to be more memory efficient

    This commit will change LIC4 to use an array of size clusterSize to keep track of unique quadrants of coordinates in the moving window of coordinates. This instead of using an ArrayList of precomputed coordinates and a HashSet for checking for number of unique coordinates in window

commit 8605e213027a727a9d4b913f61e60c7fc4510e2b
Merge: 73c6bb7 7474a57
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Wed Jan 29 17:24:14 2025 +0100

    Merge pull request #35 from VincentLagerros/Feture/issue#15

    feat: Added LIC10, LIC11 and tests

commit 15c4b30d81305bd2d60dce987bd35ffe4b99f3f9
Merge: 23faf9c 73c6bb7
Author: Julia <75080978+Juliapp123@users.noreply.github.com>
Date:   Wed Jan 29 17:22:56 2025 +0100

    Merge branch 'main' into LIC1

commit 23faf9c1ebd92f936e3affc2388f45342ca4b08e
Author: Julia Potrus <juliapo@kth.se>
Date:   Wed Jan 29 17:17:53 2025 +0100

    feat: add LIC1 and connected tests

commit 73c6bb786f00ea700d05d0a703cf322282bb16f2
Merge: bfe99cb 5834072
Author: ErikBetzholtz <97543780+ErikBetzholtz@users.noreply.github.com>
Date:   Wed Jan 29 16:41:27 2025 +0100

    Merge pull request #40 from VincentLagerros/lic9bugfix

    Bugfix: fix LIC9 by changing

commit 5834072b7ced9aec42e97f8e5bbfdfa120e12123
Author: Vincent <vincent.lagerros@gmail.com>
Date:   Wed Jan 29 16:20:34 2025 +0100

    Bugfix: fix LIC9 by changing

    Fixes Consecutive data points interpretation

commit 7474a574df8ace04a1c73bb60148414c9d584780
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 12:47:05 2025 +0100

    Fix: added input testing to LIC 11

commit e848cddab9933883f39f91b422c594c59caf16b4
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 12:37:57 2025 +0100

    Fix: Added input testing to LIC 10

commit 7e1ce822977f471e5b111ce228efebe1c4df8287
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 11:00:38 2025 +0100

    refactor: changed name of a test for LIC11

commit 2f655f65b24a4c2689c2a2c2965fd9f2f1dbe892
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 10:56:53 2025 +0100

    feat: Added LIC11 and tests

commit ed4650634306def3dfc1d929efabf0f7ed364b8b
Author: RasmussBrodin <rasmuss.brodin@gmail.com>
Date:   Wed Jan 29 10:08:37 2025 +0100

    feat: Added LIC10 and tests

ErikBetzholtz committed:
Merge remote-tracking branch 'origin/main' into feature/issue#18-LIC13

merge will add changes from main to feature branch for LIC13 function

VincentLagerros commented on src/main/java/se/kth/EvaluateLIC.java:
```diff
@@ -432,6 +432,90 @@ public boolean LIC11(Point2D[] coordinates, int gPts) {
+            if(condition1&&condition2) return true;
+
+            // calculate triangle area between points
+            double k = 0.5 * (Math.abs(pt1.getX() * pt2.getY() + pt2.getX() * pt3.getY() + pt3.getX() * pt1.getY() - pt1.getX() * pt3.getY() - pt2.getX() *
```
you can use the getTriangleArea function I added

ErikBetzholtz replied:
nice, ill add it

ErikBetzholtz committed:
minor: change LIC13 to using dedicated function for triangle area

will change the function LIC13 to use getTriangleArea funciton instead of duplicating code
